woocommerce/wc-admin
# Adding SlotFill support for API-driven tasks components

Fixes #7373

This PR implements SlotFill for optional use in replacing a task's appearance within a task list, as well as utilizing it for displaying all current task components once clicked on. 

Adds slot-fill components to onboarding package:
- `<WooOnboardingTaskListItem />`
- `<WooOnboardingTask />`

_Notes:_
- This involves copying over all the current task components and temporarily duplicating them (while modifying the new ones). This creates a less-than-pretty diff, but the copy/paste changes are captured in the single commit d7b1c4f. It might be best to parse through the other commits instead of looking at the global diff.
- A few things, such as body classes and other classes for styles seem a little messy. They were already used elsewhere in this feature so this might be a target of a later clean-up PR. 
- I had to move around the new logic regarding `trackedCompletedActions` for the Marketing task in commit ffa9d85d9ec7b8f5db00ab808450f83caabe9f91. @louwie17 might want to ensure this is as expected.

### Detailed test instructions:

-   Checkout branch
-   Ensure tasks feature is enabled
- Smoke test all tasks to ensure that they appear and function as they have previously
- You can test the `WooOnboardingTaskListItem` by creating and importing a test component that looks something like below. This will result in the title for the payments task changing to "Alternative payments task title."
```js
/**
 * External dependencies
 */
import { registerPlugin } from '@wordpress/plugins';
import { WooOnboardingTaskListItem } from '@woocommerce/onboarding';

registerPlugin( 'wc-admin-onboarding-slot-test', {
	scope: 'woocommerce-admin',
	render: () => (
		<WooOnboardingTaskListItem id="payments">
			{ ( { defaultTaskItem } ) => (
				<>
					{ React.cloneElement( defaultTaskItem, {
						title: 'Alternative payments task title',
					} ) }
				</>
			) }
		</WooOnboardingTaskListItem>
	),
} );
```
